lib/fastlane/plugin/secrets_manager_storage/storage.rb

```diff
@@ -213,14 +220,47 @@ def delete_secret(secret_name)
 
       private
 
+      def generate_tags_for_secret(secret_file)
+        return {} unless File.file?(secret_file)
+
+        expiry = nil
+        secret_specific_tags = {}
+        case File.extname(secret_file)
+        when ".p12"
+          # not sure how to get expiry of the cert
+        when ".cer"
+          cert_info = Match::Utils.get_cert_info(secret_file)
+          secret_specific_tags["Name"] = cert_info
+            .find { |attribute| attribute.first == "Common Name" }
+            .last
+          expiry = cert_info.find { |attribute| attribute.first == "End Datetime" }.last
+        when ".mobileprovision"
+          secret_specific_tags[
+            "Name"
+          ] = `/usr/libexec/PlistBuddy -c 'Print Name' /dev/stdin <<< $(security cms -D -i "#{secret_file}")`.chomp.strip
```

Comment:
This is not great for portability obviously, is it expected or do we need a note somewhere / handling of errors?

Reply:
match only works on Mac. no one else makes it this convoluted :)